### PR TITLE
Add backend files for question service and add .gitignore

### PR DESCRIPTION
**Pull request by jjtoh**

### CODE REVIEW

File: backend/QuestionService/src/controllers/questionController.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        // } else {
+        //     res.status(500).send();
+        //     console.log("Internal server error");
+        // }
+    }
+}
+
+
+export const addQuestion = async(req: Request, res: Response, next : NextFunction) => {
+    const {qid, title, description, catagories, complexity} = req.body;
```
Comment (erjunze) [inline backend/QuestionService/src/controllers/questionController.ts:41]: Typo, should be categories

File: backend/QuestionService/src/controllers/questionController.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    try {
+        const question = await Question.create({qid, title, description, catagories, complexity});
```
Comment (erjunze) [inline backend/QuestionService/src/controllers/questionController.ts:43]: Same here

File: backend/QuestionService/src/models/questionModel.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    title: {
+        type: String,
+        required: true,
+        unique: true
+    },
+    description: {
+        type: String,
+        required: true
+    },
+    catagories: {
```
Comment (erjunze) [inline backend/QuestionService/src/models/questionModel.ts:20]: Same here

File: backend/QuestionService/src/utils/validation.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+            .notEmpty().withMessage("Missing question ID.").bail()
+            .isInt({min : 1}).withMessage("Question IDs should be integer values starting from 1.").bail()
+            .custom( async (value) => {
+                if (await Question.findOne({qid : value})) {
+                    throw new Error("Question ID already exists.")
+                }
+            }),
+        body("title").notEmpty().withMessage("Missing question title."),
+        body("description").notEmpty().withMessage("Missing question description."),
+        body("catagories").isLength({min: 1}).withMessage("Missing question catagory."),
```
Comment (erjunze) [inline backend/QuestionService/src/utils/validation.ts:17]: Same here

**Review by erjunze — COMMENTED** [review]
The typo is in the database as well, can that be fixed?
Aside from that, if this code works, what else do we need to add or change?

File: backend/QuestionService/src/utils/validation.ts (hunk 4 above)
Comment (jjtoh) [reply backend/QuestionService/src/utils/validation.ts:17]: My bad, will be fixed in next commit...

**Review by Seonlo99 — APPROVED** [review]
LGTM

**Pull request merged**